test/Microsoft.AspNet.Mvc.Core.Test/Routing/InnerAttributeRouteTest.cs

```diff
@@ -361,6 +361,63 @@ public class InnerAttributeRouteTest
         }
 
         [Theory]
+        [InlineData("template", "template/{parameter}", "/template/5", true)]
+        [InlineData("template", "template/{parameter}", "/template", false)]
+        [InlineData("template/{parameter}", "template/{parameter}/{id}", "/template/5/1234", true)]
+        [InlineData("template", "template/{parameter:int=1033}", "/template", false)]
+        [InlineData("template", "template/{parameter:int=5}", "/template/5", true)]
+        public void AttributeRoute_GenerateLink_UseAvailableVariables(
+            string firstTemplate,
+            string secondTemplate,
+            string expectedPath,
+            bool useValues)
+        {
+            // Arrange
+            var expectedGroup = CreateRouteGroup(0, useValues ? secondTemplate : firstTemplate);
```

Comment:
When you have conditional statements in the arrange section of your tests it usually indicates the need to break out the test (even if it means duplicating lots of code).